test/util/SignatureLibTest.js

```diff
@@ -0,0 +1,176 @@
+// JS Libraries
+const withData = require('leche').withData;
+const { t } = require('../utils/consts');
+const { hashInterestRequest, hashInterestResponse, hashLoanTermsRequest, hashLoanTermsResponse } = require('../utils/hashes');
+const { createInterestRequest, createUnsignedInterestResponse, createLoanRequest, createUnsignedLoanResponse } = require('../utils/structs');
+const ethUtil = require('ethereumjs-util')
+
+// Smart contracts
+const Mock = artifacts.require("./mock/util/Mock.sol");
+const SignatureLibMock = artifacts.require("./mock/util/SignatureLibMock.sol");
+
+// constants
+const { NULL_ADDRESS } = require('../utils/consts');
+const chains = require('../utils/chains');
+
+contract('SignatureLibTest', function (accounts) {
+    const lendersAddress = accounts[3];
+    const loansAddress = accounts[4];
+    let consensusInstance;
+    let instance;
+
+    beforeEach('Setup for each test', async () => {
+        instance = await SignatureLibMock.new();
+        consensusInstance = await Mock.new();
+    });
+
+    withData({
+        _1_first_test_hashInterestRequest: [chains.mainnet, accounts[2], 234764, 344673177, 34467317723],
```

Comment:
Just a suggestion: As we have a number already in the name and we will split this test file into smaller files, I don't think we need to clarify **_first_test_** or **_second_test_** on each case.

Reply:
I agree that it was a little unclear. I've renamed the tests to reflect it more accurately